### PR TITLE
Handle channelName

### LINKED ISSUE #170: Operator-Controller resolution considers spec.channel
**Goal:**
The Operator Controller should be updated to feed channel constraints into Deppy.

**Background:**
- When working on this ticket, it may be fruitful to review  [this PR](https://github.com/operator-framework/operator-controller/pull/142), which updated the operator API to allow users to specify a `spec.version` constraint.

**Acceptance Criteria:**

- The "Demo Script" outlined in #163 runs as expected
- Reasonable testing is introduced.

### PR DESCRIPTION
**Pull request by tmshort**
Adds backend support for channelName

Fixes #170
Fixes OPRUN-2917

### CODE REVIEW

**Comment by tmshort** [comment]
I am a bit concerned about item 6 in #163:
> 6. Edit the Operator object and change the channel to strimzi-0.33.x

This channel name doesn't match what was merged in #171, so I might need to update the regex.

**Comment by grokspawn** [comment]
> I am a bit concerned about item 6 in #163:
> 
> > 6. Edit the Operator object and change the channel to strimzi-0.33.x
> 
> This channel name doesn't match what was merged in #171, so I might need to update the regex.

It looks like opm would have an issue with the regex w.r.t. semver catalog template generating minor version channels: https://olm.operatorframework.io/docs/reference/catalog-templates/#example-1 (actually, the last output block).

**Review by joelanford — COMMENTED** [review]
Just a few (hopefully small) suggestions, but I think this looks good overall.

I opened a follow-up issue (https://github.com/operator-framework/operator-controller/issues/176) that captures (as best I can) the topic I brought up in the WG meeting about avoiding putting channel metadata into bundle entities. That's definitely out of scope for this milestone though, so I don't want to distract from the task at hand.

File: controllers/operator_controller_test.go
Comment (tmshort) [reply controllers/operator_controller_test.go:604]: It's easy enough to copy/paste the test w/o the package version.

File: controllers/operator_controller_test.go
Comment (joelanford) [reply controllers/operator_controller_test.go:668]: Based on how `notFoundError()` appears to work currently, the version aspect of the error only shows up when the version is set. I figure we can do a similar thing for channels. 

We should handle both being set as well and try to come up with a coherent message in that case, but again, not _super_ concerned with that in this particular PR. Do it if it's easy, but if handling it seems hard or contentious, then add a TODO comment and GH issue and move on without it. My two cents at least.

File: controllers/operator_controller_test.go
Comment (joelanford) [inline controllers/operator_controller_test.go:667]: Can we also fetch the expected Bundle Deployment and check the `.spec.source.image.ref`?

File: controllers/operator_controller_test.go
Comment (tmshort) [reply controllers/operator_controller_test.go]: done

**Comment by tmshort** [comment]
The channel name regex is now this:
```
^[a-z0-9]+([\.-][a-z0-9]+)*$
```
Which out to allow for the channel names, but might be too flexible?

**Review by perdasilva — APPROVED** [review]
Nice one, thank you!!

**Comment by perdasilva** [comment]
@tmshort maybe wait on @joelanford or @grokspawn to sign off before merging? Just in case I've missed something =S

**Comment by tmshort** [comment]
@joelanford @grokspawn you good with this?
I grabbed the demo script from @perdasilva and modified it to work with channel names per #163

**Review by joelanford — APPROVED** [review]
🚀

**Pull request merged**
